core/templates/pages/exploration-player-page/services/command-executor.service.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A service that maintains a record of the state of the player,
+ *  like engine service.
+ */
+
+
+require('services/contextual/window-ref.service.ts');
+require(
+  'pages/exploration-player-page/services/window-wrapper-message.service');
+require('./window-wrapper-message.service');
+require(
+  'pages/exploration-player-page/services/current-interaction.service.ts');
+angular.module('oppia').factory('CommandExecutorService', [
+  'WindowWrapperMessageService',
+  function(WindowWrapperMessageService) {
+    var hostname = '';
+    var cachedOuterFrameMessage = '';
+    var initialized = false;
+    var currentInteraction = '';
+    var sendParentReadyState = function() {
+      WindowWrapperMessageService.postMessageToParent(
+        'Ready to receive hostname', '*');
+    };
+    var initialize = function() {
+      WindowWrapperMessageService.addEventListener(
+        'message', (event) => {
+          var messageArray = event.data.split(' ');
```

Comment:
If you are passing structured data, consider passing in a JSON-encoded list/etc. and then JSON-decoding it here and extracting the values directly, rather than doing lots of string manipulation.